internal/template/parse.go

```
@@ -47,30 +50,46 @@ func (p *TemplateParser) Parse(name, content string) (*template.Template, error)
 	return parsed, nil
 }
 
-// AddFunction adds a custom function to the function map
+// AddFunction adds a custom function to the function map.
 func (p *TemplateParser) AddFunction(name string, fn interface{}) {
+	if isBlockedTemplateFunction(name) && !(p.allowEnvAccess && isEnvironmentTemplateFunction(name)) {
```

golangci-lint (2.12.2)

[error] QF1001: could apply De Morgan's law

(staticcheck)